src/main/java/Duke.java

```diff
@@ -1,10 +1,293 @@
+        }
+        System.out.println("You have " + tasks.size() + " task(s) on the list.\n");
+    }
+
+    private static void printWelcomeMessage() {
+        System.out.println("____________________________________________________________\n"
+                + "Hello, nice to meet you. I'm Yae! (*^▽^*)\n"
+                + "What can I do for you?\n"
+                + "____________________________________________________________");
+    }
+    
+    private static String extractDescription(String[] words) throws MissingDescriptionException {
+        if (words.length < 2) {
+            throw new MissingDescriptionException();
+        }
+        String[] parameters = words[1].split(" /", 2);
+        if (parameters[0].isBlank()) {
+            throw new MissingDescriptionException();
+        }
+        return parameters[0];
+    }
+
+    private static String extractDate(String[] words) throws MissingDateException {
+        String[] parameters = words[1].split(" /", 2);
+        parameters = parameters[1].split(" ", 2);
+        if (parameters.length < 2 || parameters[1].isBlank()) {
+            throw new MissingDateException();
+        }
+        return parameters[1];
+    }
+
+    private static void printHelp() {
+        System.out.println("Here are the list of commands:");
+        System.out.println("list: Lists current tasks.");
+        System.out.println("mark: Marks task as done. (e.g. mark <task number>)\n"
+                + "unmark: Marks task as not done. (e.g. unmark <task number>)");
+        System.out.println("todo: adds a todo. (e.g. todo <description>)");
+        System.out.println("deadline: adds a deadline. (e.g. deadline <description> /by <date>)");
+        System.out.println("event: adds an event. (e.g. event <description> /at <date>)");
+        System.out.println("delete: delete a task. (e.g. delete <task number>)");
+    }
+
+    private static void readSaveData() throws FileNotFoundException {
+        String taskType;
+        String taskDescription;
+        boolean isDone;
+        String date = "";
+        File loadData = new File("data/duke.txt");
+        Scanner loadDataScanner = new Scanner(loadData);
+        while (loadDataScanner.hasNext()) {
+            String line = loadDataScanner.nextLine();
+            String[] words = line.split(" \\| ");
+            taskType = words[0];
+            isDone = words[1].equals("[X]");
+            taskDescription = words[2];
+            if (words.length > 3) {
+                date = words[3];
+            }
+            loadData(taskType, isDone, taskDescription, date);
+        }
+        System.out.println("Loaded Save File");
+    }
+
+    private static void loadData(String command, boolean isDone, String description, String date) {
+        switch(command) {
+        case "todo":
+            tasks.add(new ToDo(description));
+            if (isDone) {
+                tasks.get(tasks.size() - 1).setDone();
+            }
+            break;
```

Comment:
Consider extracting this chunk out into a separate method for SLAP.

Same issue for the other `case` blocks.